From 58e7f390041d5578f594327ff539aaa358fbb333 Mon Sep 17 00:00:00 2001
From: Lei Xu <lei.xu@nxp.com>
Date: Thu, 12 Jun 2025 09:29:00 +0800
Subject: [PATCH 26/56] LF-15662: arm64: dts: frdm-imx93: add a new dts file
 for uart test

Add a new dts file for uart test on FRDM-IMX93 board.
We multiplex the lpuart5 and lpuart6 test ports from the GPIO pins
of P11 interface on FRDM-IMX93 board.

Signed-off-by: Lei Xu <lei.xu@nxp.com>
Reviewed-by: Sherry Sun <sherry.sun@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |  2 +-
 .../dts/freescale/imx93-11x11-frdm-lpuart.dts | 45 +++++++++++++++++++
 2 files changed, 46 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx93-11x11-frdm-lpuart.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 34f08974c73c..5e47331dd312 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -496,7 +496,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-evk.dtb \
 
 dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-frdm.dtb imx93-11x11-frdm-8mic-reve.dtb \
 			  imx93-11x11-frdm-aud-hat.dtb imx93-11x11-frdm-ld.dtb \
-			  imx93-11x11-frdm-mt9m114.dtb
+			  imx93-11x11-frdm-mt9m114.dtb imx93-11x11-frdm-lpuart.dtb
 
 imx93-11x11-frdm-ov5640-dtbs := imx93-11x11-frdm.dtb imx93-11x11-frdm-ov5640.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-frdm-ov5640.dtb
diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-lpuart.dts b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-lpuart.dts
new file mode 100644
index 000000000000..61369a29c420
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-lpuart.dts
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include "imx93-11x11-frdm.dts"
+
+/* Disable tpm3 node due to pin conflict with lpuart6. */
+/delete-node/ &tpm3;
+
+/* Unbind lpuart5 from Bluetooth, use it for lpuart test. */
+&lpuart5 {
+	/delete-node/ bluetooth;
+};
+
+&lpuart6 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart6>;
+	status = "okay";
+};
+
+/*
+ * uart test port1. Note: don't use Bluetooth at the sametime.
+ * P11: Pin27--TX Pin28--RX Pin3--CTS Pin5--RTS
+ */
+&pinctrl_uart5 {
+	fsl,pins = <
+		MX93_PAD_GPIO_IO00__LPUART5_TX      0x31e
+		MX93_PAD_GPIO_IO01__LPUART5_RX      0x31e
+		MX93_PAD_GPIO_IO02__LPUART5_CTS_B   0x31e
+		MX93_PAD_GPIO_IO03__LPUART5_RTS_B   0x31e
+	>;
+};
+
+&iomuxc {
+/* uart test port2: P11: Pin7--TX Pin29--RX Pin31--CTS Pin26--RTS */
+	pinctrl_uart6: uart6grp {
+		fsl,pins = <
+			MX93_PAD_GPIO_IO04__LPUART6_TX		0x31e
+			MX93_PAD_GPIO_IO05__LPUART6_RX		0x31e
+			MX93_PAD_GPIO_IO06__LPUART6_CTS_B	0x31e
+			MX93_PAD_GPIO_IO07__LPUART6_RTS_B	0x31e
+		>;
+	};
+};
-- 
2.34.1

